From 49dbacfbdc734a34c620cf420164a492279a3caf Mon Sep 17 00:00:00 2001
From: Cyprian Lech <clech@plume.com>
Date: Tue, 15 Sep 2020 07:02:54 +0000
Subject: [PATCH 6/7] lnx_netlink: VIF ifaces get IP in Wifi_Inet_State

This patch fixes getting IPs for vifs from Wifi_Inet_Config.
---
 src/lib/osn/src/linux/lnx_netlink.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/lib/osn/src/linux/lnx_netlink.c b/src/lib/osn/src/linux/lnx_netlink.c
index f7a7b25..b092ea3 100644
--- a/src/lib/osn/src/linux/lnx_netlink.c
+++ b/src/lib/osn/src/linux/lnx_netlink.c
@@ -101,6 +101,12 @@ bool lnx_netlink_start(lnx_netlink_t *self)
     self->nl_active = true;
     ds_dlist_insert_tail(&lnx_netlink_list, self);
 
+    /*
+     * Immediately dispatch an event -- this ensures that at least 1 event is
+     * received by the listener and that it is received sooner rather than later
+     */
+    lnx_netlink_dispatch(self->nl_events, self->nl_ifname);
+
 exit:
     return self->nl_active;
 }
-- 
2.25.1

